app/connectors/BusinessRegistrationConnector.scala

```diff
@@ -38,6 +39,8 @@ trait BusinessRegistrationConnector {
 
   def retrieveBusinessProfile(implicit hc: HeaderCarrier, rds: HttpReads[BusinessProfile]): Future[BusinessProfile] = {
     http.GET[BusinessProfile](s"$businessRegUrl/business-registration/business-tax-registration") recover {
+      case _ : NotFoundException =>
+        throw new BRDocumentNotFound("[CurrentProfileService] [getRegIdAndStatus] Could not find a BR document")
```

Comment:
could call logResponse before throwing the exception

Reply:
Is it an "error" though?